src/kernel/queue.rs

```diff
@@ -24,12 +26,14 @@ pub fn queue<Msg: Message>() -> (QueueWriter<Msg>, QueueReader<Msg>) {
 
 #[derive(Clone)]
 pub struct QueueWriter<Msg: Message> {
-    tx: Sender<Envelope<Msg>>,
+    tx: Arc<Mutex<Sender<Envelope<Msg>>>>,
```

Comment:
@vlad9486 
just nore, if we use `SyncSender` here, we dont need Arc/Mutex,
but SyncSender cannot be unbounded